storage/s3.go

```diff
@@ -0,0 +1,91 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+type S3FileStorage struct {
+	bucket string
+	client *s3.Client
+}
+
+func NewS3FileStorage(
+	bucket string, region string, apiKey string, apiSecret string,
+) (*S3FileStorage, error) {
```

Comment:
実はこう書けます

```suggestion
func NewS3FileStorage(
	bucket, region, apiKey, apiSecret string,
) (*S3FileStorage, error) {
```

とは言え、同じ型のパラメータが並ぶと絶対に取り違えるので、ここはstructを渡すようにして欲しいです。いっそのこと`LoadDefaultConfig`を外で呼ぶことを想定して、`aws.Config`を受け取るようにすると良さそうです